app/controllers/fines_controller.rb

```diff
@@ -5,15 +5,36 @@ class FinesController < ApplicationController
   before_action :authenticate_user!
 
   def index
-    @fines = if params[:group]
-               patron.group_fines
-             else
-               patron.fines
-             end
-    @checkouts = if params[:group]
-                   patron.group_checkouts.sort_by(&:due_date)
-                 else
-                   patron.checkouts.sort_by(&:due_date)
-                 end
+    @fines = fines
+    @checkouts = checkouts
+    @payments = if payments
+                  Array.wrap(payments).map { |payment| Payment.new(payment) }
+                else
+                  {}
```

Comment:
So now that we're using `Array.wrap` and ensuring that we're returning an array, we should make the fallback return object an array too (instead of a hash here)